### PR TITLE
[JVM IR] Refactor InterfaceDelegationPhase

### PR DESCRIPTION
**Pull request by kandersen**
Rename and refactor interface delegation phase. It didn't actually do
what was said on the box. It should be more in line with the
intentions of the phase now.

### CODE REVIEW

**Comment by kandersen** [comment]
Small drive-by touch up as I try to understand where calls to DefaultImpls are generated.

File: compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InheritedDefaultMethodsOnClassesLowering.kt [hunk 1]
```diff
@@ -30,20 +30,22 @@ import org.jetbrains.kotlin.ir.declarations.IrSimpleFunction
...
+import org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid
+import org.jetbrains.kotlin.ir.visitors.acceptChildrenVoid
+import org.jetbrains.kotlin.ir.visitors.transformChildrenVoid
 import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-internal val interfaceDelegationPhase = makeIrFilePhase(
-    ::InterfaceDelegationLowering,
-    name = "InterfaceDelegation",
-    description = "Delegate calls to interface members with default implementations to DefaultImpls"
+internal val inheritedDefaultMethodsOnClassesPhases = makeIrFilePhase(
```
Comment (udalov) [inline compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InheritedDefaultMethodsOnClassesLowering.kt]: phases -> phase?

File: compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/BridgeLowering.kt [hunk 2]
```diff
@@ -487,7 +484,7 @@ private data class SignatureWithSource(val signature: Method, val source: IrSimp
 
 
 fun IrSimpleFunction.overriddenInClasses(): Sequence<IrSimpleFunction> =
-    allOverridden().filter { !(it.parent.safeAs<IrClass>()?.isInterface ?: true) }
+    allOverridden().filter { !it.hasInterfaceParent() }
```
Comment (udalov) [inline compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/BridgeLowering.kt]: Consider `filterNot(IrDeclaration::hasInterfaceParent)`

File: compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InheritedDefaultMethodsOnClassesLowering.kt (hunk 1 above)
Comment (kandersen) [reply compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InheritedDefaultMethodsOnClassesLowering.kt]: Yes, thank you!

File: compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/BridgeLowering.kt (hunk 2 above)
Comment (kandersen) [reply compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/BridgeLowering.kt]: Good, thanks!

**Comment by kandersen** [comment]
Also noticed I could promote it to a ClassLoweringPass, saves an override :)

**Comment by udalov** [comment]
Thanks!

**Comment by kandersen** [comment]
cc @sfs FYI :)

**Pull request merged**